### PR TITLE
KM-18222: Add migration flow to the tvOS PlatformSDK tunnel

### PR DESCRIPTION
**Pull request by kp-diego-trevisan**
Migrating to the PlatformSDK tunnel removed the legacy IKEv2 configuration on the spot, disconnecting anyone who was connected with no warning and no reconnect. It now runs behind a "VPN Upgrade" screen, shown only when there is a live tunnel to tear down, and reconnects afterwards. Consent replaces `usePlatformSDKVPN` as the gate on tvOS.

### CODE REVIEW

**Comment by kp-diego-trevisan** [comment]
> No CRITICAL or HIGH issues. The migration flow is well-factored (consent gate replacing the flag, NE-status read instead of the onboarding flag, reconnect keyed on the reading taken before bootstrap) and the test coverage is genuinely thorough — including the timeout, double-tap, and late-arrival paths.
> 
> One MEDIUM edge case, not blocking:
> 
> `PIA VPN-tvOS/PlatformSDKMigration/Domain/Use cases/PlatformSDKMigrationUseCase.swift`
> 
> * **MEDIUM** — In the relaunch path where consent is already on record but cleanup never latched (`didConfirmPlatformSDKMigration == true`, `didCleanupLegacyVPNProfiles == false`, e.g. a prior `removeAll` returned `false`), if the `isVPNConnected` read exceeds `vpnStatusTimeout` (default 3s), the timeout answers `false` → `bootstrapApp()` runs `cleanupLegacyVPNProfilesIfNeeded()` synchronously while `wasConnectedBeforeMigration` is still `false`. A user who was live on the legacy tunnel then gets its config removed with no reconnect. This is the mirror of `testShouldConfirmMigration_WhenTheVPNStatusArrivesAfterTheTimeout`, which only passes because the test delivers the late reading _before_ invoking cleanup; in the real synchronous bootstrap sequence the reading has not landed yet. Narrow (requires a prior cleanup failure plus a >3s NE read), and the default first-launch path stays safely on legacy, so not blocking.

It's fine, users are informed before the procedure that there will be a drop and they consent to it. In this edge case they will just manually connect.

File: PIA VPN-tvOS/PlatformSDKMigration/Domain/Interfaces/LegacyVPNProfilesType.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
+//
+//  LegacyVPNProfilesType.swift
+//  PIA VPN-tvOS
+//
+//  Copyright © 2026 Private Internet Access Inc. All rights reserved.
+//
+
+import Foundation
+
+protocol LegacyVPNProfilesType {
```
Comment (kp-mario-nachbaur) [inline PIA VPN-tvOS/PlatformSDKMigration/Domain/Interfaces/LegacyVPNProfilesType.swift]: Consider making these new protocols `Sendable` to ease swift 6 migration.

**Review by kp-mario-nachbaur — DISMISSED** [review]
Looks good. Just a small suggestion.

**Review by kp-mario-nachbaur — APPROVED** [review]
